### PR TITLE
mysql prepared statements

### PR DESCRIPTION
**Pull request by Lagovas**
* support prepared statements with [binary](https://dev.mysql.com/doc/internals/en/binary-protocol-resultset.html) and [text](https://dev.mysql.com/doc/internals/en/com-query-response.html#packet-ProtocolText::Resultset) formats of response
* test support of prepared statements with postgresql/mysql (two formats)
* fix race condition (when one goroutine close connection with secure session and second try to write/encrypt)
* increase wait time in acra-censor tests for correct running on slow machines (need refactoring in a future)
* small refactoring

### CODE REVIEW

File: cmd/acra-server/listener.go [hunk 1]
```diff
@@ -402,6 +410,11 @@ func (server *SServer) StartCommands() {
 func (server *SServer) StartCommandsFromFileDescriptor(fd uintptr) {
 	var connection net.Conn
 	file := os.NewFile(fd, "/tmp/acra-server_http_api")
+	if file == nil {
+		log.Errorln("can't create new file from descriptor for api listener")
```
Comment (vixentael) [inline cmd/acra-server/listener.go]: usually we start logs with capital letter :)

just saying

File: cmd/utils.go [hunk 2]
```diff
@@ -58,10 +58,10 @@ func (handler *SignalHandler) AddCallback(callback SignalCallback) {
 
 // Register should be called as goroutine
 func (handler *SignalHandler) Register() {
-	for _, osSignal := range handler.signals {
-		signal.Notify(handler.ch, osSignal)
-	}
+	signal.Notify(handler.ch, handler.signals...)
```
Comment (vixentael) [inline cmd/utils.go]: so it notifies about many signals, right?

File: decryptor/mysql/response_proxy.go [hunk 3]
```diff
@@ -325,7 +326,28 @@ func (handler *MysqlHandler) processBinaryDataRow(rowData []byte, fields []*Colu
...
+	// no data in response
+	if rowData[0] == EOF_PACKET {
+		return rowData, nil
+	}
+
+	if rowData[0] != OK_PACKET {
+		return nil, ErrMalformPacket
+	}
+
+	pos = 1 + ((len(fields) + 7 + 2) >> 3)
```
Comment (vixentael) [inline decryptor/mysql/response_proxy.go]: 7? 2?

can you please add some comment or save this variables as constants?

File: cmd/utils.go (hunk 2 above)
Comment (Lagovas) [reply cmd/utils.go]: yes

File: cmd/acra-server/listener.go (hunk 1 above)
Comment (Lagovas) [reply cmd/acra-server/listener.go]: will update. forgot

**Pull request merged**